internal/builtinfunctions/functions.go

```diff
@@ -525,3 +528,76 @@ func getReadFileFunction() schema.CallableFunction {
 	}
 	return funcSchema
 }
+
+// CombinedObjPropertyConstantName is the identifier string key that points to the container of constant or repeated values.
+const CombinedObjPropertyConstantName = "constant"
+
+// CombinedObjPropertyItemName is the identifier string key that points to the container of a list of objects.
+const CombinedObjPropertyItemName = "item"
+
+// CombinedObjIDDelimiter is the delimiter for joining Object ID strings and/or TypeID strings.
+const CombinedObjIDDelimiter = "__"
+
+func getBindConstantsFunction() schema.CallableFunction {
+	funcSchema, err := schema.NewDynamicCallableFunction(
+		"bindConstants",
+		[]schema.Type{
+			schema.NewListSchema(schema.NewAnySchema(), nil, nil),
+			schema.NewAnySchema()},
+		schema.NewDisplayValue(
+			schema.PointerTo("Bind Constants"),
+			schema.PointerTo(
+				"Creates a list of objects with ID `CombinedObject`. "+
+					fmt.Sprintf("Each object has two properties `%s` and `%s`.\n", CombinedObjPropertyItemName, CombinedObjPropertyConstantName)+
+					fmt.Sprintf("Param 1: Value(s) to be included in the `%s` field \n", CombinedObjPropertyItemName)+
+					fmt.Sprintf("Param 2: Value(s) to populate the field `%s` with every output item", CombinedObjPropertyConstantName)),
+			nil),
+		func(items []any, columnValues any) (any, error) {
+			combinedItems := make([]any, len(items))
+			for k, itemValue := range items {
+				combinedItems[k] = map[string]any{
+					CombinedObjPropertyItemName:     itemValue,
+					CombinedObjPropertyConstantName: columnValues,
+				}
+			}
+			return combinedItems, nil
+		},
+		HandleTypeSchemaCombine,
+	)
+	if err != nil {
+		panic(err)
+	}
+	return funcSchema
+}
+
+// HandleTypeSchemaCombine returns the type that is output by the 'bindConstants' function.
+// Its parameter list requires a ListSchema and one other schema of any type.
+// A new ListSchema of ObjectSchemas is created from the two given types.
+func HandleTypeSchemaCombine(inputType []schema.Type) (schema.Type, error) {
+	if len(inputType) != 2 {
+		return nil, fmt.Errorf("expected exactly two types")
+	}
+	itemsType, isList := inputType[0].(*schema.ListSchema)
+	if !isList {
+		return nil, fmt.Errorf("expected first type to be a list schema")
+	}
+	itemType := itemsType.ItemsValue
+	constantsTypeArg := inputType[1]
+	combinedObjectName := schemaName(itemType) + CombinedObjIDDelimiter + schemaName(constantsTypeArg)
+	return schema.NewListSchema(
+		schema.NewObjectSchema(
+			combinedObjectName,
+			map[string]*schema.PropertySchema{
+				CombinedObjPropertyItemName:     schema.NewPropertySchema(itemType, nil, false, nil, nil, nil, nil, nil),
+				CombinedObjPropertyConstantName: schema.NewPropertySchema(constantsTypeArg, nil, false, nil, nil, nil, nil, nil),
```

Comment:
because `inputType[0]` is not reused, and `inputType[1]` was reused

one could do this, but it felt unnecessary
```go
itemsTypeArg := inputType[0]
itemsType, isList := itemsTypeArg.(*schema.ListSchema)
```